src/disco/events/fd_circbuf.h

```diff
@@ -0,0 +1,65 @@
+#ifndef HEADER_fd_src_disco_events_fd_circbuf_h
+#define HEADER_fd_src_disco_events_fd_circbuf_h
+
+#include "../fd_disco_base.h"
+
+#define FD_CIRCBUF_ALIGN (4096UL)
+
+struct __attribute__((aligned(FD_CIRCBUF_ALIGN))) fd_circbuf_private {
+  ulong cnt;
+  ulong head;
+  ulong tail;
```

Comment:
I think it's worth adding some comments. These are the offsets relative to the end of this struct of the first and last message, respectively. In data structures like this, tail is often a pointer to after the end of the last message, so it's worth noting that this is to the start of the last message